kotlinx-coroutines-core/common/src/flow/SharedFlow.kt

```diff
@@ -708,7 +720,32 @@ internal open class SharedFlowImpl<T>(
 
     override fun fuse(context: CoroutineContext, capacity: Int, onBufferOverflow: BufferOverflow) =
         fuseSharedFlow(context, capacity, onBufferOverflow)
-    
+
+
+    /* Entities used by IDEA debugger agent */
+
+    private class FlowValueWrapper<T>(val value: T)
+
+    @Suppress("UNCHECKED_CAST")
+    private fun <T> wrap(value: T): T = value
+    @Suppress("UNCHECKED_CAST")
+    private fun <T> unwrap(value: T): T = value
+
+    // debugger agent substitutes a call to this function to `wrap`
+    private fun wrapDebuggerCapture(value: Any?): Any? {
+        val wrapper = FlowValueWrapper(value)
+        debuggerCapture(wrapper)
+        return wrapper
+    }
+
+    // debugger agent substitutes a call to this function to `unwrap`
+    private fun unwrapDebuggerCapture(value: Any?) = (value as FlowValueWrapper<*>).value
```

Comment:
it seems like in some rare cases value could be not FlowValueWrapper instance, so it may be better to add ? here
For example if we attach agent and some values are now wrapped already